### PR TITLE
Add ObservableMap

### PR DESCRIPTION
**Pull request by katis**
Added tests, combined ObservableList and ObservableMap into the same module so that list could use the hidden `_wrap` constructor of the map to implement `asMap()`.

Still missing `observe`

### CODE REVIEW

File: mobx/lib/src/api/observable_collections.dart [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import 'dart:collection';
+import 'dart:math';
+
+import 'package:meta/meta.dart';
+import 'package:mobx/mobx.dart';
+import 'package:mobx/src/core.dart';
+
+part 'observable_collections/observable_list.dart';
```
Comment (pavanpodila) [inline mobx/lib/src/api/observable_collections.dart]: like the naming (`observable_collections`)

**Comment by pavanpodila** [comment]
coverage needs to be improved though

**Comment by katis** [comment]
Coverage back to positive, merging

**Comment by katis** [comment]
I hadn't merged this for some reason 😕

**Comment by pavanpodila** [comment]
Merging for now...we can add the `observe` in the next PR

**Pull request merged**